DaoBuilder/daoUtils.ts

```diff
@@ -0,0 +1,193 @@
+  LinearERC20Voting as ILinearERC20Voting,
+  KeyValuePairs as IKeyValuePairs,
+  FractalRegistry as IFractalRegistry,
+  GnosisSafe,
+  GnosisSafeProxyFactory,
+} from "@fractal-framework/fractal-contracts";
+import { getCreate2Address, solidityKeccak256 } from "ethers/lib/utils";
+import {
+  getMultiSendCallOnlyDeployment,
+  getProxyFactoryDeployment,
+  getSafeSingletonDeployment,
+} from "@safe-global/safe-deployments";
+const { AddressZero, HashZero } = ethers.constants;
+
+export const CHAIN_ID = 5;
+export const SAFE_VERSION = "1.3.0";
+
+/* eslint-disable node/no-unsupported-features/es-syntax */
+async function getFractalContractAddressByChainId(_chainId: number) {
+  const networkPath = _chainId === 1 ? "mainnet" : "goerli";
+
+  const Azorius = await import(
+    `@fractal-framework/fractal-contracts/deployments/${networkPath}/Azorius.json`
+  );
+  const FractalRegistry = await import(
+    `@fractal-framework/fractal-contracts/deployments/${networkPath}/FractalRegistry.json`
+  );
+  const KeyValuePairs = await import(
+    `@fractal-framework/fractal-contracts/deployments/${networkPath}/KeyValuePairs.json`
+  );
+  const LinearERC20Voting = await import(
+    `@fractal-framework/fractal-contracts/deployments/${networkPath}/LinearERC20Voting.json`
+  );
+  const ModuleProxyFactory = await import(
+    `@fractal-framework/fractal-contracts/deployments/${networkPath}/ModuleProxyFactory.json`
+  );
+  return {
+    Azorius,
+    FractalRegistry,
+    KeyValuePairs,
+    LinearERC20Voting,
+    ModuleProxyFactory,
+  };
+}
+/* eslint-enable node/no-unsupported-features/es-syntax */
+
+export const getMasterCopies = async (): Promise<{
+  zodiacModuleProxyFactoryContract: IModuleProxyFractory;
+  fractalAzoriusMasterCopyContract: IAzorius;
+  fractalRegistryContract: IFractalRegistry;
+  linearVotingMasterCopyContract: ILinearERC20Voting;
+  keyValuePairContract: IKeyValuePairs;
+  multisendContract: Contract;
+}> => {
+  const {
+    Azorius,
+    FractalRegistry,
+    KeyValuePairs,
+    LinearERC20Voting,
+    ModuleProxyFactory,
+  } = await getFractalContractAddressByChainId(CHAIN_ID);
+
+  const zodiacModuleProxyFactoryContract = (await ethers.getContractAt(
+    ModuleProxyFactory.abi as Record<string, any>[],
+    ModuleProxyFactory.address
+  )) as IModuleProxyFractory;
+
+  const fractalAzoriusMasterCopyContract = (await ethers.getContractAt(
+    Azorius.abi,
+    Azorius.address
+  )) as IAzorius;
+
+  const linearVotingMasterCopyContract = (await ethers.getContractAt(
+    LinearERC20Voting.abi,
+    LinearERC20Voting.address
+  )) as ILinearERC20Voting;
+
+  const fractalRegistryContract = (await ethers.getContractAt(
+    FractalRegistry.abi as Record<string, any>[],
+    FractalRegistry.address
+  )) as IFractalRegistry;
+
+  const keyValuePairContract = (await ethers.getContractAt(
+    KeyValuePairs.abi as Record<string, any>[],
+    KeyValuePairs.address
+  )) as IKeyValuePairs;
```

Comment:
I'm going to note here so that its tested (or a test written) but three of these had a fit in my ide when I added the dynamic importer. might need to check and make sure there isn't any weird about the mainnet deployments for these.